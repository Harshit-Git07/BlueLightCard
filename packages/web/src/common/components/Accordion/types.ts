export interface AccordionProps {
  title: string;
  children: React.ReactNode;
  onClickOpen?: () => void;
}
