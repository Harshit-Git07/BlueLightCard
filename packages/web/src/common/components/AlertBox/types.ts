export interface AlertBoxProps {
  alertType: string;
  title: string;
  description: any;
}

export interface style {
  textColor: string;
  backgroundColor: string;
}

export interface config {
  [key: string]: style;
}
