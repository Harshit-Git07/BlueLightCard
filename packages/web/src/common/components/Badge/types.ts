export type BgColorString = `bg-${string}`;

export type BadgeProps = {
  label: string;
  color: BgColorString;
  size: 'small' | 'large';
};
