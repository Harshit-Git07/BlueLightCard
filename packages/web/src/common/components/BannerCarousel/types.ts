import { CampaignCardProps } from '../CampaignCard/types';

export type BannerCarouselPropers = {
  banners: CampaignCardProps[];
};
