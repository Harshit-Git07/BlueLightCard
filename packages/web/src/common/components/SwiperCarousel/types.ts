export type CarouselProps = {
  children: React.ReactNode[] | React.ReactNode;

  // Display
  hideArrows?: boolean;
  hidePillButtons?: boolean;
  loop?: boolean;

  // Breakpoints
  elementsPerPageDesktop?: number;
  elementsPerPageLaptop?: number;
  elementsPerPageTablet?: number;
  elementsPerPageMobile?: number;

  // Auto play
  autoPlay?: boolean;
  autoPlayIntervalMs?: number;
};
