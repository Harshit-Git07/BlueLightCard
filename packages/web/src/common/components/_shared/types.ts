import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ChangeEventHandler, ForwardedRef, MouseEventHandler, PropsWithChildren } from 'react';

export type InputFieldWrapperProps = PropsWithChildren & {
  icon?: IconDefinition;
  iconRight?: IconDefinition;
  showRightIcon?: boolean;
  showErrorState?: boolean;
  select?: boolean;
  showSuccessState?: boolean;
  onRightIconClick?: MouseEventHandler;
};

export interface InputFieldSharedProps<E, V = string> {
  name?: string;
  error?: boolean;
  success?: boolean;
  value?: V;
  required?: boolean;
  _ref?: ForwardedRef<unknown>;
  onChange?: ChangeEventHandler<E>;
}
