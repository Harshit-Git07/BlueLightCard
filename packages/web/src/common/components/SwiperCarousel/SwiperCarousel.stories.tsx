import { Meta, StoryFn } from '@storybook/react';
import SwiperCarousel from './SwiperCarousel';

const componentMeta: Meta<typeof SwiperCarousel> = {
  title: 'Component System/SwiperCarousel',
  component: SwiperCarousel,
};

const DefaultTemplate: StoryFn = (args) => (
  <SwiperCarousel {...args} elementsPerPageLaptop={5}>
    <div className="bg-red-500 w-16 h-16">item 1</div>
    <div className="bg-red-500 w-16 h-16">item 2</div>
    <div className="bg-red-500 w-16 h-16">item 3</div>
    <div className="bg-red-500 w-16 h-16">item 4</div>
    <div className="bg-red-500 w-16 h-16">item 5</div>
  </SwiperCarousel>
);

export const Default = DefaultTemplate.bind({});

Default.args = {
  hidePillButtons: false,
  loop: false,
  autoPlay: false,
  hideArrows: false,
};

export const hiddenArrows = DefaultTemplate.bind({});

hiddenArrows.args = {
  hidePillButtons: false,
  loop: true,
  autoPlay: false,
  hideArrows: true,
};

export const autoLoop = DefaultTemplate.bind({});

autoLoop.args = {
  hidePillButtons: false,
  loop: true,
  autoPlay: true,
  hideArrows: false,
};

export const autoSlide = DefaultTemplate.bind({});

autoSlide.args = {
  hidePillButtons: false,
  loop: false,
  autoPlay: true,
  hideArrows: false,
};

export const hiddenPills = DefaultTemplate.bind({});

hiddenPills.args = {
  hidePillButtons: true,
  loop: true,
  autoPlay: false,
  hideArrows: false,
};

export const hiddenPillsAuto = DefaultTemplate.bind({});

hiddenPillsAuto.args = {
  hidePillButtons: true,
  loop: true,
  autoPlay: true,
  hideArrows: false,
};

export default componentMeta;
