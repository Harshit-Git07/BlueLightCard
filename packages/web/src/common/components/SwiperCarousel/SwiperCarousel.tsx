import React from 'react';

import { Pagination, Navigation, Autoplay } from 'swiper/modules';
import { Swiper, SwiperSlide } from 'swiper/react';

import { CarouselProps } from './types';

import 'swiper/css';
import 'swiper/css/pagination';
import 'swiper/css/navigation';

const SwiperCarousel: React.FC<CarouselProps> = ({
  children,
  elementsPerPageDesktop = 5,
  elementsPerPageLaptop = 5,
  elementsPerPageTablet = 3,
  elementsPerPageMobile = 1,
  navigation,
  pagination,
}) => {
  return (
    <Swiper
      modules={[Navigation, Pagination, Autoplay]}
      loop
      autoplay={{ delay: 5000, disableOnInteraction: false, pauseOnMouseEnter: true }}
      spaceBetween={5}
      slidesPerView={elementsPerPageMobile}
      pagination={pagination ? { dynamicBullets: true, clickable: true } : undefined}
      navigation={navigation ?? undefined}
      grabCursor={true}
      breakpoints={{
        // when window width is >= 768px
        768: {
          slidesPerView: elementsPerPageTablet,
          spaceBetween: 10,
        },
        // when window width is >= 1024px
        1024: {
          slidesPerView: elementsPerPageLaptop,
          spaceBetween: 15,
        },
        // when window with is >= 1200
        1200: {
          slidesPerView: elementsPerPageDesktop,
          spaceBetween: 20,
        },
      }}
      className={pagination ? 'swiper-pagination-enabled' : ''}
    >
      {React.Children.map(children, (child, index) => (
        <SwiperSlide key={index}>{child}</SwiperSlide>
      ))}
    </Swiper>
  );
};

export default SwiperCarousel;
