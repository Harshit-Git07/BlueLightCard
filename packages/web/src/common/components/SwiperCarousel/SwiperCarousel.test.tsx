import SwiperCarousel from '@/components/SwiperCarousel/SwiperCarousel';
import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import { CarouselProps } from './types';

jest.mock('../../hooks/useIsDarkMode', () => ({
  __esModule: true,
  default: () => true,
}));

jest.mock('swiper/react', () => ({
  Swiper: ({ children }: CarouselProps) => <div data-testid="swiper-testid">{children}</div>,
  SwiperSlide: ({ children }: CarouselProps) => (
    <div data-testid="swiper-slide-testid">{children}</div>
  ),
}));

jest.mock('swiper/modules', () => ({
  Navigation: () => null,
  Pagination: () => null,
  Autoplay: () => null,
}));

jest.mock('swiper/css', () => jest.fn());

describe('SwiperCarousel component', () => {
  const props = {
    elementsPerPageDesktop: 5,
    elementsPerPageLaptop: 3,
    elementsPerPageTablet: 2,
    elementsPerPageMobile: 1,
    autoPlay: true,
    autoPlayIntervalMs: 5000,
  };

  it('renders without crashing', () => {
    render(
      <SwiperCarousel {...props}>
        <div>Item</div>
        <div>Item</div>
        <div>Item</div>
        <div>Item</div>
        <div>Item</div>
        <div>Item</div>
      </SwiperCarousel>
    );
  });

  it('matches snapshot', () => {
    const component = renderer.create(
      <SwiperCarousel {...props}>
        <div>Item</div>
        <div>Item</div>
        <div>Item</div>
      </SwiperCarousel>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays correct number of elements on screen size', () => {
    render(
      <SwiperCarousel {...props}>
        <div>Item</div>
        <div>Item</div>
        <div>Item</div>
      </SwiperCarousel>
    );
    const slides = screen.getAllByTestId('swiper-slide-testid');
    expect(slides.length).toBe(3);
  });
});
