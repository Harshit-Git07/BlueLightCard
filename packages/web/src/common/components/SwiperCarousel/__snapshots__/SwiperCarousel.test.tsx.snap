// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwiperCarousel component matches snapshot 1`] = `
<div
  data-testid="swiper-testid"
>
  <div
    data-testid="swiper-slide-testid"
  >
    <div>
      Item
    </div>
  </div>
  <div
    data-testid="swiper-slide-testid"
  >
    <div>
      Item
    </div>
  </div>
  <div
    data-testid="swiper-slide-testid"
  >
    <div>
      Item
    </div>
  </div>
</div>
`;
